query/physicalplan/synchronize.go

```diff
@@ -62,5 +63,12 @@ func (m *Synchronizer) Draw() *Diagram {
 }
 
 func (m *Synchronizer) Close() {
-	m.next.Close()
+	if !m.closed {
```

Comment:
So I think this is fine, but could we keep this close logic consistent with the finish logic where it waits to call close until it has received all the expected close calls

Reply:
ohhh yeah sure I'll make it so 👍

Reply:
done :+1: